library/Backends/llvm/src/Juvix/Backends/LLVM/Compilation.hs

```diff
@@ -353,15 +370,57 @@ compileRecord ty (recordName, _) =
 
 compileField ::
   Types.Define m =>
-  Monad m =>
   Types.TypeLLVM ->
   -- | Primitive value.
   Types.TermFieldSelector ->
   -- | Type of the primitive.
   m LLVM.Operand
 compileField ty (recordName, fieldName, term) = do
   compiledTerm <- compileTerm term
-  Record.loadField recordName fieldName compiledTerm (typeToLLVM ty)
+  llvmType <- typeToLLVM ty
+  Record.loadField recordName fieldName compiledTerm llvmType
+
+compileSumDecl ::
+  Types.Define m =>
+  Types.SumDecl ->
+  -- | Primitive value.
+  Types.Annotated Types.TermLLVM ->
+  -- | Type of the primitive.
+  m LLVM.Operand
+compileSumDecl (sumName, variantDecls) term = do
+  llvmTypes <- mapM (typeToLLVM . snd) variantDecls
```

Comment:
Please prefer `traverse` to `mapM` due to 

```haskell
λ> traverse
traverse
  :: (Traversable t, Applicative f) => (a -> f b) -> t a -> f (t b)
λ> mapM
mapM :: (Traversable t, Monad m) => (a -> m b) -> t a -> m (t b)
```

Applicative is a smaller interface and thus a bit more general

Reply:
I personally prefer `mapM` here. I find it easier to read. I think it's fine to not always use the more general interface. I usually use `return` instead of `pure`, I prefer `map` over `fmap` for lists, I prefer `Nothing` over `mzero`, etc.